File: src/main/java/com/nao4j/subtrol/repository/UserRepository.java
package com.nao4j.subtrol.repository;

import com.nao4j.subtrol.document.User;
import com.nao4j.subtrol.dto.UserCredentials;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Optional;

public interface UserRepository extends MongoRepository<User, String> {

    Optional<UserCredentials> findUserCredentialsByEmail(String email);

}

File: src/main/kotlin/com/nao4j/subtrol/repository/UserRepository.kt
package com.nao4j.subtrol.repository

import com.nao4j.subtrol.dto.UserCredentials
import com.nao4j.subtrol.model.User
import org.springframework.data.mongodb.repository.MongoRepository

interface UserRepository : MongoRepository<User, String> {

    fun findUserCredentialsByEmail(email: String?): UserCredentials?

}
